src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,183 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public MyLinkedList() {
+    }
+
     @Override
     public void add(T value) {
+        if (head == null) {
+            linkHead(value);
+        } else {
+            linkTail(value);
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == 0) {
+            linkHead(value);
+        } else if (index == size) {
+            linkTail(value);
+        } else {
+            checkIndexToAdd(index);
+            linkBefore(value, findNode(index));
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        if (index == 0) {
+            return head.item;
+        } else if (index == size - 1) {
+            return tail.item;
+        }
```

Comment:
Do you really need these conditionals?  method findNode() also return these values